src/briefcase/integrations/subprocess.py

```diff
@@ -605,25 +608,36 @@ def _stream_output_thread(self, popen_process, filter_func):
         :param filter_func: a callable that will be invoked on every line
             of output that is streamed; see ``stream_output`` for details.
         """
-        # ValueError is raised if stdout is unexpectedly closed.
-        # This can happen if the user starts spamming CTRL+C, for instance.
-        # Silently exit to avoid Python printing the exception to the console.
-        # StopIteration can be raised by a filter function to indicate that
-        # there's no need to stream any more.
-        with suppress(ValueError, StopIteration):
+        try:
             while True:
-                # readline should always return at least a newline (ie \n)
-                # UNLESS the underlying process is exiting/gone; then "" is returned
-                output_line = ensure_str(popen_process.stdout.readline())
+                try:
+                    output_line = ensure_str(popen_process.stdout.readline())
+                except ValueError:
+                    # ValueError is raised if stdout is unexpectedly closed. This can
+                    # happen if the user starts spamming CTRL+C, for instance. Silently
+                    # exit to avoid Python printing the exception to the console.
+                    return
+
+                # readline should always return at least a newline (ie \n) UNLESS
+                # the underlying process is exiting/gone; then "" is returned.
                 if output_line:
                     if filter_func:
-                        filtered_output = filter_func(output_line.rstrip("\n"))
-                        if filtered_output is not None:
-                            self.tools.logger.info(filtered_output)
+                        try:
+                            for filtered_output in filter_func(
+                                output_line.rstrip("\n")
+                            ):
+                                self.tools.logger.info(filtered_output)
+                        except StopStreaming:
+                            return
                     else:
                         self.tools.logger.info(output_line)
                 else:
                     return
+        except Exception as e:
+            self.tools.logger.error(
+                f"Error while streaming output: {e.__class__.__name__}: {e}"
+            )
+            self.tools.logger.capture_stacktrace()
```

Comment:
Using `capture_stacktrace()` here may create confusing outcomes.
- In perhaps the best outcome, the Rich stacktrace in the log is simply from the streamer's exception....assuming the main thread doesn't raise another elsewhere.
- If the main thread raises, it becomes a bit of a race condition which stacktrace is included in the log.

Even in the current usecase for testing, if the streamer raises, then Briefcase will later raise `BriefcaseCommandError: Test suite didn't report a result`....but only one of the stracktraces will make it in the log.

I think that `capture_stacktrace()` should be reserved for main thread exceptions since they will probably be more interesting/useful.

This is a "least effort" implementation to simply add the streamer's Rich stacktrace to the bottom of the log:
```python
        except Exception as e:
            self.tools.logger.error(
                f"Error while streaming output: {e.__class__.__name__}: {e}"
            )
            from rich.traceback import Traceback
            stracktrace = [
                "Output Streamer Stacktrace:",
                Traceback(
                    width=self.tools.logger.print.LOG_FILE_WIDTH,
                    show_locals=True,
                ),
            ]
            self.tools.logger.add_log_file_extra(
                lambda: self.tools.logger.print.log.print(*stracktrace)
            )
```

Alternatively, of course, refactoring stracktrace management in `Console` to accommodate an arbitrary number of stracktraces is possible as well.

Reply:
Good call; and it wasn't *that* hard to make the stacktrace management generic. It also now allows for capturing stack traces that don't result in the app crashing - not sure how useful that will be, but it's an option at least.